libraries/AP_EFI/AP_EFI_Serial_Hirth.cpp

```diff
@@ -110,7 +110,7 @@ void AP_EFI_Serial_Hirth::check_response()
         computed_checksum += res_data.quantity = port->read();
         computed_checksum += res_data.code = port->read();
 
-        if (res_data.code == requested_code) {
+        if (res_data.code == requested_code && res_data.quantity > QUANTITY_REQUEST_STATUS) {
```

Comment:
This structure is still bad.

If this `if` returns false then we still have one change in 256 of consuming this packet as real.  We should simply discard the input in the case that `res_data.code != requested_code`

Also, shouldn't this be
```suggestion
        if (res_data.code == requested_code && res_data.quantity == expected_bytes) {
```
(possible +/- header, I haven't checked)